docs/modules/operation/pages/provisioning/detectors/BsfDetector.adoc

```diff
@@ -0,0 +1,175 @@
+= BSF Detector
+
+The BSF Detector lets you run arbitrary scripts using the Bean Scripting Framework to determine the availability of a service.
+This can be a script written in any of the languages BSF supports.
+The only requirement is that the script returns the string `OK` if it passes.
+
+Unlike the BSF Monitor, the BSF Detector only passes in the `map`, `ip_addr`, `svc_name`, and `results` variables as beans to the script.
+
+== Detector facts
+
+[options="autowidth"]
+|===
+| Implementation | `org.opennms.netmgt.provision.detector.bsf.BSFDetector`
+|===
+
+== Configuration and Use
+
+.Parameters for the BSF detector
+[options="header, autowidth"]
+
+|===
+| Parameter        | Description                                                                                            | Required | Default value
+| `fileName`       | The path to the file on the OpenNMS server which contains the BSF-compatible script to run.            | optional |
+| `langClass`      | The short name of the language the script is written in. Choices: Jython, BeanShell, Groovy, etc.     | optional | 
+| `bsfEngine`      | The BSF engine to use when running the script. Choices:
+
+* bsh.util.BeanShellBSFEngine
+* org.codehaus.groovy.bsf.GroovyEngine
+* org.apache.bsf.engines.jython.JythonEngine                                                                                | optional | 
```

Comment:
Lists in table does not work. The choices you have to use in your script, I would expect it also in `code` formatting.

<img width="836" alt="Screenshot 2021-04-19 at 09 46 13" src="https://user-images.githubusercontent.com/1095181/115200096-251c2d00-a0f4-11eb-8009-116f2ca2d33b.png">

Reply:
Now it works: https://github.com/OpenNMS/opennms/blob/82b7042d726b29f3f15001ae7c88d9283d6869b2/docs/modules/operation/pages/provisioning/detectors/BsfDetector.adoc